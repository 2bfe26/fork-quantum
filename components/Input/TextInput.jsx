import styled, { css } from 'styled-components';
import Colors from '../Colors';

const TextInput = styled.input.attrs({
  type: 'text',
})`
  box-sizing: border-box;
  margin-top: 8px;
  padding-right: 42px;
  &::-webkit-inner-spin-button,
  &::-webkit-outer-spin-button {
    -webkit-appearance: none;
    margin: 0;
  }

  background-color: ${Colors.WHITE};
  border-radius: 4px;
  border: 1.5px solid ${Colors.BLACK['400']};
  box-sizing: border-box;
  color: ${Colors.BLACK['700']};
  font-size: initial;
  height: 44px;
  min-height: 44px;
  letter-spacing: 0.2px;
  outline: none;
  padding: 10px 12px;
  transition: all 0.2s ease-in-out;
  width: 100%;

  :hover,
  :focus {
    border-color: ${Colors.BLUE['500']};
    box-shadow: 0 2px 6px 0 ${Colors.BLUE['50']};
  }

  ${({ error }) =>
    error &&
    css`
      border-color: ${Colors.ERROR['500']};

      :hover,
      :focus {
        border-color: ${Colors.ERROR['500']};
        box-shadow: 0 2px 6px 0 ${Colors.ERROR['500']};
      }
    `};

  &[disabled] {
    background-color: ${Colors.BLACK['100']};
    border-color: ${Colors.BLACK['400']};
    box-shadow: none;
    color: ${Colors.BLACK['400']};
    cursor: not-allowed;
  }

  ${({ placeholder, defaultValue, value }) =>
    placeholder &&
    !defaultValue &&
    !value &&
    `
    color: ${Colors.BLACK['400']};
  `}

  ${({ searchable }) =>
    searchable &&
    `
    padding: 10px 42px;
  `}

  ${({ password }) =>
    password &&
    `
    padding-right: 28px;
  `}

  &::-webkit-calendar-picker-indicator {
    display: none;
  }

  :-webkit-autofill {
    box-shadow: 0 0 0px 1000px ${Colors.BLUE['200']} inset;
  }
`;

TextInput.displayName = 'TextInput';

export default TextInput;
