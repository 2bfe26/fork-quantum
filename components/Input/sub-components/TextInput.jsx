import PropTypes from 'prop-types';
import styled from 'styled-components';
import { spacing, colors } from '../../shared/theme';
import { shadow } from '../../shared';

const TextInput = styled.input`
  border-radius: 4px;
  box-sizing: border-box;
  box-sizing: border-box;
  font-size: initial;
  height: 44px;
  letter-spacing: 0.2px;
  min-height: 44px;
  outline: none;
  transition: all 0.2s ease-in-out;
  width: 100%;

  &::-webkit-inner-spin-button,
  &::-webkit-outer-spin-button {
    -webkit-appearance: none;
    margin: 0;
  }

  &::-webkit-calendar-picker-indicator {
    display: none;
  }

  ${({ defaultValue, error, hasIcon, placeholder, theme, value }) => {
    const {
      colors: {
        primary: { 500: primaryColor },
        error: { 100: error100, 500: error500 },
        neutral: {
          100: neutral100,
          300: neutral300,
          500: neutral500,
          700: neutral700,
        },
      },
      spacing: { xsmall, small, xxlarge },
    } = theme;

    const mainColor = error ? error500 : primaryColor;

    return `
      background-color: ${neutral100};
      border: 2px solid ${neutral500};
      color: ${neutral700};
      margin-top: ${xsmall}px;
      padding: ${small}px;

      ${hasIcon ? `padding-right: ${xxlarge}px;` : ''}
      ${error ? `border-color: ${mainColor};` : ''}
      ${placeholder && !defaultValue && !value ? `color: ${neutral500};` : ''}

      :hover, :focus {
        border-color: ${mainColor};
        ${shadow(5, mainColor)({ theme })}
      }

      &[disabled] {
        background-color: ${neutral300};
        border-color: ${neutral500};
        box-shadow: none;
        color: ${neutral500};
        cursor: not-allowed;
      }

      :-webkit-autofill {
        box-shadow: 0 0 0 1000px ${error100} inset;
      }
    `;
  }}
`;

TextInput.displayName = 'TextInput';

TextInput.propTypes = {
  theme: PropTypes.shape({
    spacing: PropTypes.object,
    colors: PropTypes.object,
  }),
};

TextInput.defaultProps = {
  theme: { spacing, colors },
};

export default TextInput;
