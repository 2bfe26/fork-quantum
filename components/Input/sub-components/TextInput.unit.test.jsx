import React from 'react';
import renderer from 'react-test-renderer';
import TextInput from './TextInput';

describe('Input component', () => {
  it('should match snapshots', () => {
    const INPUTS = [
      <TextInput />,
      <TextInput error="Error message" />,
      <TextInput disabled />,
      <TextInput hasIcon />,
      <TextInput placeholder="Placeholder text" />,
      <TextInput value="Value text" />,
      <TextInput defaultValue="Value text" />,
      <TextInput placeholder="Placeholder text" value="Value text" />,
      <TextInput error="Error message" disabled />,
    ];

    INPUTS.forEach(input =>
      expect(renderer.create(input).toJSON()).toMatchSnapshot(),
    );
  });
});
