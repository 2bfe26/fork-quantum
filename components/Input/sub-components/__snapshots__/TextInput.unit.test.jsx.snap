// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input component should match snapshots 1`] = `
.c0 {
  -webkit-appearance: none;
  border-radius: 4px;
  box-sizing: border-box;
  box-sizing: border-box;
  font-size: initial;
  -webkit-letter-spacing: 0.2px;
  -moz-letter-spacing: 0.2px;
  -ms-letter-spacing: 0.2px;
  letter-spacing: 0.2px;
  outline: none;
  -webkit-transition: all 0.2s ease-in-out;
  transition: all 0.2s ease-in-out;
  width: 100%;
  margin-top: 0px;
  padding: 8px 16px;
  background-color: #ffffff;
  border: 2px solid #999999;
  color: #424242;
}

.c0::-webkit-inner-spin-button,
.c0::-webkit-outer-spin-button {
  -webkit-appearance: none;
  margin: 0;
}

.c0::-webkit-calendar-picker-indicator,
.c0::-webkit-search-cancel-button {
  display: none;
}

.c0::-ms-clear {
  display: none;
}

.c0:hover,
.c0:focus {
  border-color: #1250C4;
  box-shadow: 0px 3px 1px -2px rgba(18,80,196,0.2),0px 2px 2px 0px rgba(18,80,196,0.14),0px 1px 5px 0px rgba(18,80,196,0.12);
}

.c0[disabled] {
  background-color: #f2f2f2;
  border-color: #999999;
  color: #999999;
}

.c0:-webkit-autofill {
  box-shadow: 0 0 0 1000px #fff5f5 inset;
}

.c0[disabled] {
  box-shadow: none;
  cursor: not-allowed;
}

<input
  className="c0"
/>
`;

exports[`Input component should match snapshots 2`] = `
.c0 {
  -webkit-appearance: none;
  border-radius: 4px;
  box-sizing: border-box;
  box-sizing: border-box;
  font-size: initial;
  -webkit-letter-spacing: 0.2px;
  -moz-letter-spacing: 0.2px;
  -ms-letter-spacing: 0.2px;
  letter-spacing: 0.2px;
  outline: none;
  -webkit-transition: all 0.2s ease-in-out;
  transition: all 0.2s ease-in-out;
  width: 100%;
  margin-top: 0px;
  padding: 8px 16px;
  background-color: rgba(0,0,0,0.7);
  border: 2px solid transparent;
  color: #ffffff;
}

.c0::-webkit-inner-spin-button,
.c0::-webkit-outer-spin-button {
  -webkit-appearance: none;
  margin: 0;
}

.c0::-webkit-calendar-picker-indicator,
.c0::-webkit-search-cancel-button {
  display: none;
}

.c0::-ms-clear {
  display: none;
}

.c0::-webkit-input-placeholder {
  color: #ffffff;
}

.c0::-moz-placeholder {
  color: #ffffff;
}

.c0:-ms-input-placeholder {
  color: #ffffff;
}

.c0::placeholder {
  color: #ffffff;
}

.c0:hover,
.c0:focus {
  border-color: #0CC0EA;
}

.c0[disabled] {
  color: rgba(255,255,255,0.5);
  border-color: transparent;
}

.c0[disabled] {
  box-shadow: none;
  cursor: not-allowed;
}

<input
  className="c0"
/>
`;

exports[`Input component should match snapshots 3`] = `
.c0 {
  -webkit-appearance: none;
  border-radius: 4px;
  box-sizing: border-box;
  box-sizing: border-box;
  font-size: initial;
  -webkit-letter-spacing: 0.2px;
  -moz-letter-spacing: 0.2px;
  -ms-letter-spacing: 0.2px;
  letter-spacing: 0.2px;
  outline: none;
  -webkit-transition: all 0.2s ease-in-out;
  transition: all 0.2s ease-in-out;
  width: 100%;
  margin-top: 0px;
  padding: 8px 16px;
  background-color: #ffffff;
  border: 2px solid #999999;
  color: #424242;
  border-color: #e91b0c;
}

.c0::-webkit-inner-spin-button,
.c0::-webkit-outer-spin-button {
  -webkit-appearance: none;
  margin: 0;
}

.c0::-webkit-calendar-picker-indicator,
.c0::-webkit-search-cancel-button {
  display: none;
}

.c0::-ms-clear {
  display: none;
}

.c0:hover,
.c0:focus {
  border-color: #e91b0c;
  box-shadow: 0px 3px 1px -2px rgba(233,27,12,0.2),0px 2px 2px 0px rgba(233,27,12,0.14),0px 1px 5px 0px rgba(233,27,12,0.12);
}

.c0[disabled] {
  background-color: #f2f2f2;
  border-color: #999999;
  color: #999999;
}

.c0:-webkit-autofill {
  box-shadow: 0 0 0 1000px #fff5f5 inset;
}

.c0[disabled] {
  box-shadow: none;
  cursor: not-allowed;
}

<input
  className="c0"
/>
`;

exports[`Input component should match snapshots 4`] = `
.c0 {
  -webkit-appearance: none;
  border-radius: 4px;
  box-sizing: border-box;
  box-sizing: border-box;
  font-size: initial;
  -webkit-letter-spacing: 0.2px;
  -moz-letter-spacing: 0.2px;
  -ms-letter-spacing: 0.2px;
  letter-spacing: 0.2px;
  outline: none;
  -webkit-transition: all 0.2s ease-in-out;
  transition: all 0.2s ease-in-out;
  width: 100%;
  margin-top: 0px;
  padding: 8px 16px;
  background-color: rgba(0,0,0,0.7);
  border: 2px solid transparent;
  color: #ffffff;
  border-color: #ffffff;
}

.c0::-webkit-inner-spin-button,
.c0::-webkit-outer-spin-button {
  -webkit-appearance: none;
  margin: 0;
}

.c0::-webkit-calendar-picker-indicator,
.c0::-webkit-search-cancel-button {
  display: none;
}

.c0::-ms-clear {
  display: none;
}

.c0::-webkit-input-placeholder {
  color: #ffffff;
}

.c0::-moz-placeholder {
  color: #ffffff;
}

.c0:-ms-input-placeholder {
  color: #ffffff;
}

.c0::placeholder {
  color: #ffffff;
}

.c0:hover,
.c0:focus {
  border-color: #0CC0EA;
}

.c0[disabled] {
  color: rgba(255,255,255,0.5);
  border-color: transparent;
}

.c0[disabled] {
  box-shadow: none;
  cursor: not-allowed;
}

<input
  className="c0"
/>
`;

exports[`Input component should match snapshots 5`] = `
.c0 {
  -webkit-appearance: none;
  border-radius: 4px;
  box-sizing: border-box;
  box-sizing: border-box;
  font-size: initial;
  -webkit-letter-spacing: 0.2px;
  -moz-letter-spacing: 0.2px;
  -ms-letter-spacing: 0.2px;
  letter-spacing: 0.2px;
  outline: none;
  -webkit-transition: all 0.2s ease-in-out;
  transition: all 0.2s ease-in-out;
  width: 100%;
  margin-top: 0px;
  padding: 8px 16px;
  background-color: #ffffff;
  border: 2px solid #999999;
  color: #424242;
}

.c0::-webkit-inner-spin-button,
.c0::-webkit-outer-spin-button {
  -webkit-appearance: none;
  margin: 0;
}

.c0::-webkit-calendar-picker-indicator,
.c0::-webkit-search-cancel-button {
  display: none;
}

.c0::-ms-clear {
  display: none;
}

.c0:hover,
.c0:focus {
  border-color: #1250C4;
  box-shadow: 0px 3px 1px -2px rgba(18,80,196,0.2),0px 2px 2px 0px rgba(18,80,196,0.14),0px 1px 5px 0px rgba(18,80,196,0.12);
}

.c0[disabled] {
  background-color: #f2f2f2;
  border-color: #999999;
  color: #999999;
}

.c0:-webkit-autofill {
  box-shadow: 0 0 0 1000px #fff5f5 inset;
}

.c0[disabled] {
  box-shadow: none;
  cursor: not-allowed;
}

<input
  className="c0"
  disabled={true}
/>
`;

exports[`Input component should match snapshots 6`] = `
.c0 {
  -webkit-appearance: none;
  border-radius: 4px;
  box-sizing: border-box;
  box-sizing: border-box;
  font-size: initial;
  -webkit-letter-spacing: 0.2px;
  -moz-letter-spacing: 0.2px;
  -ms-letter-spacing: 0.2px;
  letter-spacing: 0.2px;
  outline: none;
  -webkit-transition: all 0.2s ease-in-out;
  transition: all 0.2s ease-in-out;
  width: 100%;
  margin-top: 0px;
  padding: 8px 16px;
  background-color: rgba(0,0,0,0.7);
  border: 2px solid transparent;
  color: #ffffff;
}

.c0::-webkit-inner-spin-button,
.c0::-webkit-outer-spin-button {
  -webkit-appearance: none;
  margin: 0;
}

.c0::-webkit-calendar-picker-indicator,
.c0::-webkit-search-cancel-button {
  display: none;
}

.c0::-ms-clear {
  display: none;
}

.c0::-webkit-input-placeholder {
  color: #ffffff;
}

.c0::-moz-placeholder {
  color: #ffffff;
}

.c0:-ms-input-placeholder {
  color: #ffffff;
}

.c0::placeholder {
  color: #ffffff;
}

.c0:hover,
.c0:focus {
  border-color: #0CC0EA;
}

.c0[disabled] {
  color: rgba(255,255,255,0.5);
  border-color: transparent;
}

.c0[disabled] {
  box-shadow: none;
  cursor: not-allowed;
}

<input
  className="c0"
  disabled={true}
/>
`;

exports[`Input component should match snapshots 7`] = `
.c0 {
  -webkit-appearance: none;
  border-radius: 4px;
  box-sizing: border-box;
  box-sizing: border-box;
  font-size: initial;
  -webkit-letter-spacing: 0.2px;
  -moz-letter-spacing: 0.2px;
  -ms-letter-spacing: 0.2px;
  letter-spacing: 0.2px;
  outline: none;
  -webkit-transition: all 0.2s ease-in-out;
  transition: all 0.2s ease-in-out;
  width: 100%;
  margin-top: 0px;
  padding: 8px 16px;
  background-color: #ffffff;
  border: 2px solid #999999;
  color: #424242;
}

.c0::-webkit-inner-spin-button,
.c0::-webkit-outer-spin-button {
  -webkit-appearance: none;
  margin: 0;
}

.c0::-webkit-calendar-picker-indicator,
.c0::-webkit-search-cancel-button {
  display: none;
}

.c0::-ms-clear {
  display: none;
}

.c0:hover,
.c0:focus {
  border-color: #1250C4;
  box-shadow: 0px 3px 1px -2px rgba(18,80,196,0.2),0px 2px 2px 0px rgba(18,80,196,0.14),0px 1px 5px 0px rgba(18,80,196,0.12);
}

.c0[disabled] {
  background-color: #f2f2f2;
  border-color: #999999;
  color: #999999;
}

.c0:-webkit-autofill {
  box-shadow: 0 0 0 1000px #fff5f5 inset;
}

.c0[disabled] {
  box-shadow: none;
  cursor: not-allowed;
}

<input
  className="c0"
/>
`;

exports[`Input component should match snapshots 8`] = `
.c0 {
  -webkit-appearance: none;
  border-radius: 4px;
  box-sizing: border-box;
  box-sizing: border-box;
  font-size: initial;
  -webkit-letter-spacing: 0.2px;
  -moz-letter-spacing: 0.2px;
  -ms-letter-spacing: 0.2px;
  letter-spacing: 0.2px;
  outline: none;
  -webkit-transition: all 0.2s ease-in-out;
  transition: all 0.2s ease-in-out;
  width: 100%;
  margin-top: 0px;
  padding: 8px 16px;
  background-color: rgba(0,0,0,0.7);
  border: 2px solid transparent;
  color: #ffffff;
}

.c0::-webkit-inner-spin-button,
.c0::-webkit-outer-spin-button {
  -webkit-appearance: none;
  margin: 0;
}

.c0::-webkit-calendar-picker-indicator,
.c0::-webkit-search-cancel-button {
  display: none;
}

.c0::-ms-clear {
  display: none;
}

.c0::-webkit-input-placeholder {
  color: #ffffff;
}

.c0::-moz-placeholder {
  color: #ffffff;
}

.c0:-ms-input-placeholder {
  color: #ffffff;
}

.c0::placeholder {
  color: #ffffff;
}

.c0:hover,
.c0:focus {
  border-color: #0CC0EA;
}

.c0[disabled] {
  color: rgba(255,255,255,0.5);
  border-color: transparent;
}

.c0[disabled] {
  box-shadow: none;
  cursor: not-allowed;
}

<input
  className="c0"
/>
`;

exports[`Input component should match snapshots 9`] = `
.c0 {
  -webkit-appearance: none;
  border-radius: 4px;
  box-sizing: border-box;
  box-sizing: border-box;
  font-size: initial;
  -webkit-letter-spacing: 0.2px;
  -moz-letter-spacing: 0.2px;
  -ms-letter-spacing: 0.2px;
  letter-spacing: 0.2px;
  outline: none;
  -webkit-transition: all 0.2s ease-in-out;
  transition: all 0.2s ease-in-out;
  width: 100%;
  margin-top: 0px;
  padding: 8px 16px;
  background-color: #ffffff;
  border: 2px solid #999999;
  color: #424242;
}

.c0::-webkit-inner-spin-button,
.c0::-webkit-outer-spin-button {
  -webkit-appearance: none;
  margin: 0;
}

.c0::-webkit-calendar-picker-indicator,
.c0::-webkit-search-cancel-button {
  display: none;
}

.c0::-ms-clear {
  display: none;
}

.c0:hover,
.c0:focus {
  border-color: #1250C4;
  box-shadow: 0px 3px 1px -2px rgba(18,80,196,0.2),0px 2px 2px 0px rgba(18,80,196,0.14),0px 1px 5px 0px rgba(18,80,196,0.12);
}

.c0[disabled] {
  background-color: #f2f2f2;
  border-color: #999999;
  color: #999999;
}

.c0:-webkit-autofill {
  box-shadow: 0 0 0 1000px #fff5f5 inset;
}

.c0[disabled] {
  box-shadow: none;
  cursor: not-allowed;
}

<input
  className="c0"
  placeholder="Placeholder text"
/>
`;

exports[`Input component should match snapshots 10`] = `
.c0 {
  -webkit-appearance: none;
  border-radius: 4px;
  box-sizing: border-box;
  box-sizing: border-box;
  font-size: initial;
  -webkit-letter-spacing: 0.2px;
  -moz-letter-spacing: 0.2px;
  -ms-letter-spacing: 0.2px;
  letter-spacing: 0.2px;
  outline: none;
  -webkit-transition: all 0.2s ease-in-out;
  transition: all 0.2s ease-in-out;
  width: 100%;
  margin-top: 0px;
  padding: 8px 16px;
  background-color: rgba(0,0,0,0.7);
  border: 2px solid transparent;
  color: #ffffff;
}

.c0::-webkit-inner-spin-button,
.c0::-webkit-outer-spin-button {
  -webkit-appearance: none;
  margin: 0;
}

.c0::-webkit-calendar-picker-indicator,
.c0::-webkit-search-cancel-button {
  display: none;
}

.c0::-ms-clear {
  display: none;
}

.c0::-webkit-input-placeholder {
  color: #ffffff;
}

.c0::-moz-placeholder {
  color: #ffffff;
}

.c0:-ms-input-placeholder {
  color: #ffffff;
}

.c0::placeholder {
  color: #ffffff;
}

.c0:hover,
.c0:focus {
  border-color: #0CC0EA;
}

.c0[disabled] {
  color: rgba(255,255,255,0.5);
  border-color: transparent;
}

.c0[disabled] {
  box-shadow: none;
  cursor: not-allowed;
}

<input
  className="c0"
  placeholder="Placeholder text"
/>
`;

exports[`Input component should match snapshots 11`] = `
.c0 {
  -webkit-appearance: none;
  border-radius: 4px;
  box-sizing: border-box;
  box-sizing: border-box;
  font-size: initial;
  -webkit-letter-spacing: 0.2px;
  -moz-letter-spacing: 0.2px;
  -ms-letter-spacing: 0.2px;
  letter-spacing: 0.2px;
  outline: none;
  -webkit-transition: all 0.2s ease-in-out;
  transition: all 0.2s ease-in-out;
  width: 100%;
  margin-top: 0px;
  padding: 8px 16px;
  background-color: #ffffff;
  border: 2px solid #999999;
  color: #424242;
}

.c0::-webkit-inner-spin-button,
.c0::-webkit-outer-spin-button {
  -webkit-appearance: none;
  margin: 0;
}

.c0::-webkit-calendar-picker-indicator,
.c0::-webkit-search-cancel-button {
  display: none;
}

.c0::-ms-clear {
  display: none;
}

.c0:hover,
.c0:focus {
  border-color: #1250C4;
  box-shadow: 0px 3px 1px -2px rgba(18,80,196,0.2),0px 2px 2px 0px rgba(18,80,196,0.14),0px 1px 5px 0px rgba(18,80,196,0.12);
}

.c0[disabled] {
  background-color: #f2f2f2;
  border-color: #999999;
  color: #999999;
}

.c0:-webkit-autofill {
  box-shadow: 0 0 0 1000px #fff5f5 inset;
}

.c0[disabled] {
  box-shadow: none;
  cursor: not-allowed;
}

<input
  className="c0"
  value="Value text"
/>
`;

exports[`Input component should match snapshots 12`] = `
.c0 {
  -webkit-appearance: none;
  border-radius: 4px;
  box-sizing: border-box;
  box-sizing: border-box;
  font-size: initial;
  -webkit-letter-spacing: 0.2px;
  -moz-letter-spacing: 0.2px;
  -ms-letter-spacing: 0.2px;
  letter-spacing: 0.2px;
  outline: none;
  -webkit-transition: all 0.2s ease-in-out;
  transition: all 0.2s ease-in-out;
  width: 100%;
  margin-top: 0px;
  padding: 8px 16px;
  background-color: rgba(0,0,0,0.7);
  border: 2px solid transparent;
  color: #ffffff;
}

.c0::-webkit-inner-spin-button,
.c0::-webkit-outer-spin-button {
  -webkit-appearance: none;
  margin: 0;
}

.c0::-webkit-calendar-picker-indicator,
.c0::-webkit-search-cancel-button {
  display: none;
}

.c0::-ms-clear {
  display: none;
}

.c0::-webkit-input-placeholder {
  color: #ffffff;
}

.c0::-moz-placeholder {
  color: #ffffff;
}

.c0:-ms-input-placeholder {
  color: #ffffff;
}

.c0::placeholder {
  color: #ffffff;
}

.c0:hover,
.c0:focus {
  border-color: #0CC0EA;
}

.c0[disabled] {
  color: rgba(255,255,255,0.5);
  border-color: transparent;
}

.c0[disabled] {
  box-shadow: none;
  cursor: not-allowed;
}

<input
  className="c0"
  value="Value text"
/>
`;

exports[`Input component should match snapshots 13`] = `
.c0 {
  -webkit-appearance: none;
  border-radius: 4px;
  box-sizing: border-box;
  box-sizing: border-box;
  font-size: initial;
  -webkit-letter-spacing: 0.2px;
  -moz-letter-spacing: 0.2px;
  -ms-letter-spacing: 0.2px;
  letter-spacing: 0.2px;
  outline: none;
  -webkit-transition: all 0.2s ease-in-out;
  transition: all 0.2s ease-in-out;
  width: 100%;
  margin-top: 0px;
  padding: 8px 16px;
  background-color: #ffffff;
  border: 2px solid #999999;
  color: #424242;
}

.c0::-webkit-inner-spin-button,
.c0::-webkit-outer-spin-button {
  -webkit-appearance: none;
  margin: 0;
}

.c0::-webkit-calendar-picker-indicator,
.c0::-webkit-search-cancel-button {
  display: none;
}

.c0::-ms-clear {
  display: none;
}

.c0:hover,
.c0:focus {
  border-color: #1250C4;
  box-shadow: 0px 3px 1px -2px rgba(18,80,196,0.2),0px 2px 2px 0px rgba(18,80,196,0.14),0px 1px 5px 0px rgba(18,80,196,0.12);
}

.c0[disabled] {
  background-color: #f2f2f2;
  border-color: #999999;
  color: #999999;
}

.c0:-webkit-autofill {
  box-shadow: 0 0 0 1000px #fff5f5 inset;
}

.c0[disabled] {
  box-shadow: none;
  cursor: not-allowed;
}

<input
  className="c0"
  defaultValue="Value text"
/>
`;

exports[`Input component should match snapshots 14`] = `
.c0 {
  -webkit-appearance: none;
  border-radius: 4px;
  box-sizing: border-box;
  box-sizing: border-box;
  font-size: initial;
  -webkit-letter-spacing: 0.2px;
  -moz-letter-spacing: 0.2px;
  -ms-letter-spacing: 0.2px;
  letter-spacing: 0.2px;
  outline: none;
  -webkit-transition: all 0.2s ease-in-out;
  transition: all 0.2s ease-in-out;
  width: 100%;
  margin-top: 0px;
  padding: 8px 16px;
  background-color: rgba(0,0,0,0.7);
  border: 2px solid transparent;
  color: #ffffff;
}

.c0::-webkit-inner-spin-button,
.c0::-webkit-outer-spin-button {
  -webkit-appearance: none;
  margin: 0;
}

.c0::-webkit-calendar-picker-indicator,
.c0::-webkit-search-cancel-button {
  display: none;
}

.c0::-ms-clear {
  display: none;
}

.c0::-webkit-input-placeholder {
  color: #ffffff;
}

.c0::-moz-placeholder {
  color: #ffffff;
}

.c0:-ms-input-placeholder {
  color: #ffffff;
}

.c0::placeholder {
  color: #ffffff;
}

.c0:hover,
.c0:focus {
  border-color: #0CC0EA;
}

.c0[disabled] {
  color: rgba(255,255,255,0.5);
  border-color: transparent;
}

.c0[disabled] {
  box-shadow: none;
  cursor: not-allowed;
}

<input
  className="c0"
  defaultValue="Value text"
/>
`;

exports[`Input component should match snapshots 15`] = `
.c0 {
  -webkit-appearance: none;
  border-radius: 4px;
  box-sizing: border-box;
  box-sizing: border-box;
  font-size: initial;
  -webkit-letter-spacing: 0.2px;
  -moz-letter-spacing: 0.2px;
  -ms-letter-spacing: 0.2px;
  letter-spacing: 0.2px;
  outline: none;
  -webkit-transition: all 0.2s ease-in-out;
  transition: all 0.2s ease-in-out;
  width: 100%;
  margin-top: 0px;
  padding: 8px 16px;
  background-color: #ffffff;
  border: 2px solid #999999;
  color: #424242;
}

.c0::-webkit-inner-spin-button,
.c0::-webkit-outer-spin-button {
  -webkit-appearance: none;
  margin: 0;
}

.c0::-webkit-calendar-picker-indicator,
.c0::-webkit-search-cancel-button {
  display: none;
}

.c0::-ms-clear {
  display: none;
}

.c0:hover,
.c0:focus {
  border-color: #1250C4;
  box-shadow: 0px 3px 1px -2px rgba(18,80,196,0.2),0px 2px 2px 0px rgba(18,80,196,0.14),0px 1px 5px 0px rgba(18,80,196,0.12);
}

.c0[disabled] {
  background-color: #f2f2f2;
  border-color: #999999;
  color: #999999;
}

.c0:-webkit-autofill {
  box-shadow: 0 0 0 1000px #fff5f5 inset;
}

.c0[disabled] {
  box-shadow: none;
  cursor: not-allowed;
}

<input
  className="c0"
  placeholder="Placeholder text"
  value="Value text"
/>
`;

exports[`Input component should match snapshots 16`] = `
.c0 {
  -webkit-appearance: none;
  border-radius: 4px;
  box-sizing: border-box;
  box-sizing: border-box;
  font-size: initial;
  -webkit-letter-spacing: 0.2px;
  -moz-letter-spacing: 0.2px;
  -ms-letter-spacing: 0.2px;
  letter-spacing: 0.2px;
  outline: none;
  -webkit-transition: all 0.2s ease-in-out;
  transition: all 0.2s ease-in-out;
  width: 100%;
  margin-top: 0px;
  padding: 8px 16px;
  background-color: rgba(0,0,0,0.7);
  border: 2px solid transparent;
  color: #ffffff;
}

.c0::-webkit-inner-spin-button,
.c0::-webkit-outer-spin-button {
  -webkit-appearance: none;
  margin: 0;
}

.c0::-webkit-calendar-picker-indicator,
.c0::-webkit-search-cancel-button {
  display: none;
}

.c0::-ms-clear {
  display: none;
}

.c0::-webkit-input-placeholder {
  color: #ffffff;
}

.c0::-moz-placeholder {
  color: #ffffff;
}

.c0:-ms-input-placeholder {
  color: #ffffff;
}

.c0::placeholder {
  color: #ffffff;
}

.c0:hover,
.c0:focus {
  border-color: #0CC0EA;
}

.c0[disabled] {
  color: rgba(255,255,255,0.5);
  border-color: transparent;
}

.c0[disabled] {
  box-shadow: none;
  cursor: not-allowed;
}

<input
  className="c0"
  placeholder="Placeholder text"
  value="Value text"
/>
`;

exports[`Input component should match snapshots 17`] = `
.c0 {
  -webkit-appearance: none;
  border-radius: 4px;
  box-sizing: border-box;
  box-sizing: border-box;
  font-size: initial;
  -webkit-letter-spacing: 0.2px;
  -moz-letter-spacing: 0.2px;
  -ms-letter-spacing: 0.2px;
  letter-spacing: 0.2px;
  outline: none;
  -webkit-transition: all 0.2s ease-in-out;
  transition: all 0.2s ease-in-out;
  width: 100%;
  margin-top: 0px;
  padding: 8px 16px;
  background-color: #ffffff;
  border: 2px solid #999999;
  color: #424242;
  border-color: #e91b0c;
}

.c0::-webkit-inner-spin-button,
.c0::-webkit-outer-spin-button {
  -webkit-appearance: none;
  margin: 0;
}

.c0::-webkit-calendar-picker-indicator,
.c0::-webkit-search-cancel-button {
  display: none;
}

.c0::-ms-clear {
  display: none;
}

.c0:hover,
.c0:focus {
  border-color: #e91b0c;
  box-shadow: 0px 3px 1px -2px rgba(233,27,12,0.2),0px 2px 2px 0px rgba(233,27,12,0.14),0px 1px 5px 0px rgba(233,27,12,0.12);
}

.c0[disabled] {
  background-color: #f2f2f2;
  border-color: #999999;
  color: #999999;
}

.c0:-webkit-autofill {
  box-shadow: 0 0 0 1000px #fff5f5 inset;
}

.c0[disabled] {
  box-shadow: none;
  cursor: not-allowed;
}

<input
  className="c0"
  disabled={true}
/>
`;

exports[`Input component should match snapshots 18`] = `
.c0 {
  -webkit-appearance: none;
  border-radius: 4px;
  box-sizing: border-box;
  box-sizing: border-box;
  font-size: initial;
  -webkit-letter-spacing: 0.2px;
  -moz-letter-spacing: 0.2px;
  -ms-letter-spacing: 0.2px;
  letter-spacing: 0.2px;
  outline: none;
  -webkit-transition: all 0.2s ease-in-out;
  transition: all 0.2s ease-in-out;
  width: 100%;
  margin-top: 0px;
  padding: 8px 16px;
  background-color: rgba(0,0,0,0.7);
  border: 2px solid transparent;
  color: #ffffff;
  border-color: #ffffff;
}

.c0::-webkit-inner-spin-button,
.c0::-webkit-outer-spin-button {
  -webkit-appearance: none;
  margin: 0;
}

.c0::-webkit-calendar-picker-indicator,
.c0::-webkit-search-cancel-button {
  display: none;
}

.c0::-ms-clear {
  display: none;
}

.c0::-webkit-input-placeholder {
  color: #ffffff;
}

.c0::-moz-placeholder {
  color: #ffffff;
}

.c0:-ms-input-placeholder {
  color: #ffffff;
}

.c0::placeholder {
  color: #ffffff;
}

.c0:hover,
.c0:focus {
  border-color: #0CC0EA;
}

.c0[disabled] {
  color: rgba(255,255,255,0.5);
  border-color: transparent;
}

.c0[disabled] {
  box-shadow: none;
  cursor: not-allowed;
}

<input
  className="c0"
  disabled={true}
/>
`;
